LEGO1/lego/legoomni/src/common/legogamestate.cpp

```diff
@@ -256,16 +256,175 @@ void LegoGameState::SerializePlayersInfo(MxS16)
+		RemoveFromWorld(*g_isleScript, 0x471, *g_isleScript, 0);
+		RemoveFromWorld(*g_isleScript, 0x472, *g_isleScript, 0);
+		RemoveFromWorld(*g_isleScript, 0x12, *g_isleScript, 0);
+		break;
+	case 0xb:
+		RemoveFromWorld(*g_isleScript, 0x47a, *g_isleScript, 0);
+		RemoveFromWorld(*g_isleScript, 0x47b, *g_isleScript, 0);
+		RemoveFromWorld(*g_isleScript, 0x47c, *g_isleScript, 0);
+		RemoveFromWorld(*g_isleScript, 0x47d, *g_isleScript, 0);
+		break;
+	case 0xc:
+		InvokeAction(Extra::e_stop, *g_regbookScript, 0, NULL);
+		InvokeAction(Extra::e_close, *g_regbookScript, 0, NULL);
+		break;
+	case 0xd:
+		InvokeAction(Extra::e_stop, *g_infoscorScript, 0, NULL);
+		InvokeAction(Extra::e_close, *g_infoscorScript, 0, NULL);
+		break;
+	case 0xe:
+		InvokeAction(Extra::e_stop, *g_jetraceScript, 0, NULL);
+		InvokeAction(Extra::e_close, *g_jetraceScript, 0, NULL);
+		InvokeAction(Extra::e_close, *g_jetracerScript, 0, NULL);
+		break;
+	case 0x12:
+		InvokeAction(Extra::e_stop, *g_carraceScript, 0, NULL);
+		InvokeAction(Extra::e_close, *g_carraceScript, 0, NULL);
+		InvokeAction(Extra::e_close, *g_carracerScript, 0, NULL);
+		break;
+	case 0x1a:
+		Lego()->RemoveWorld(*g_garageScript, 0);
+		InvokeAction(Extra::e_stop, *g_garageScript, 0, NULL);
+		InvokeAction(Extra::e_close, *g_garageScript, 0, NULL);
+		break;
+	case 0x1b:
+		RemoveFromWorld(*g_isleScript, 0x489, *g_isleScript, 0);
+		RemoveFromWorld(*g_isleScript, 0x48a, *g_isleScript, 0);
+		RemoveFromWorld(*g_isleScript, 0x48b, *g_isleScript, 0);
+		RemoveFromWorld(*g_isleScript, 0x48c, *g_isleScript, 0);
+		break;
+	case 0x1e:
+		InvokeAction(Extra::e_stop, *g_hospitalScript, 0, NULL);
+		InvokeAction(Extra::e_close, *g_hospitalScript, 0, NULL);
+		break;
+	case 0x22:
+		InvokeAction(Extra::e_stop, *g_policeScript, 0, NULL);
+		InvokeAction(Extra::e_close, *g_policeScript, 0, NULL);
+		break;
+	case 0x23:
+		RemoveFromWorld(*g_isleScript, 0x47f, *g_isleScript, 0);
+		RemoveFromWorld(*g_isleScript, 0x480, *g_isleScript, 0);
+		RemoveFromWorld(*g_isleScript, 0x481, *g_isleScript, 0);
+		RemoveFromWorld(*g_isleScript, 0x482, *g_isleScript, 0);
+		break;
+	case 0x24:
+		InvokeAction(Extra::e_stop, *g_jukeboxScript, 0x2f, NULL);
+		InvokeAction(Extra::e_stop, *g_copterScript, 0, NULL);
+		InvokeAction(Extra::e_close, *g_copterScript, 0, NULL);
+		break;
+	case 0x25:
+		InvokeAction(Extra::e_stop, *g_jukeboxScript, 0x31, NULL);
+		InvokeAction(Extra::e_stop, *g_dunecarScript, 0, NULL);
+		InvokeAction(Extra::e_close, *g_dunecarScript, 0, NULL);
+		break;
+	case 0x26:
+		InvokeAction(Extra::e_stop, *g_jukeboxScript, 0x33, NULL);
+		InvokeAction(Extra::e_stop, *g_jetskiScript, 0, NULL);
+		InvokeAction(Extra::e_close, *g_jetskiScript, 0, NULL);
+		break;
+	case 0x27:
+		InvokeAction(Extra::e_stop, *g_jukeboxScript, 0x35, NULL);
+		InvokeAction(Extra::e_stop, *g_racecarScript, 0, NULL);
+		InvokeAction(Extra::e_close, *g_racecarScript, 0, NULL);
+		break;
+	case 0x2e:
+		if (m_unk0x424 != 2) {
```

Comment:
My guess is m_unk0x424 has to do specifically with seeing whether the game has reached Act 2 or not